platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collectors.java

```diff
@@ -201,13 +206,21 @@ public Collector<T> absentIgnoring() {
 
         @Override
         public Value<Void> collectEntries(ValueConsumer consumer, ValueCollector<T> collector, ImmutableCollection.Builder<T> collection) {
-            collector.addAll(value, collection);
+            if (sideEffect != null) {
+                collector.addAll(Value.withSideEffect(value, sideEffect).get(), collection);
```

Comment:
:x: this doesn't look correct - you should not run side effect at this point (which `get` does) but return it as part of the result Value instead.

It should be something like
```java
collector.addAll(value, collection);
return Value.present().withSideEffect(sideEffect);  // deals with null internally
```

Reply:
Oh, I think I understand now, how this works, changed it